### PR TITLE
fixes logging sample and readme

### LINKED ISSUE #503: Cannot surpress logging
config.properties['hazelcast.logging'] = 'off';

Does not stop default logger on 3.12.1

### PR DESCRIPTION
**Pull request by gulcesirvanci**
Fixes #503

### CODE REVIEW

**Comment by gulcesirvanci** [comment]
Fixes #503

File: README.md [hunk 1]
```diff
@@ -3342,18 +3346,44 @@ var winstonAdapter = {
...
         'warn',
         'info',
         'debug',
         'silly'
     ],
 
-    log: function (level, className, message, furtherInfo) {
-        this.logger.log(this.levels[level], className + ' ' + message);
+    off: function (level, objectName, message, furtherInfo) {
+        this.logger.log(LogLevel.OFF, objectName, message, furtherInfo);
```
Comment (mdumandag) [inline README.md]: like the comment above, `off` is not a required method for the `ILogger interface`

File: README.md [hunk 2]
```diff
@@ -3342,18 +3346,44 @@ var winstonAdapter = {
     }),
 
     levels: [
+        'off',
```
Comment (mdumandag) [inline README.md]: off is not a valid `LogLevel`

File: code_samples/logging.js [hunk 3]
```diff
@@ -19,59 +19,55 @@ var Config = require('hazelcast-client').Config;
...
-            levels: [
-                'error',
-                'warn',
-                'info',
-                'debug',
-                'silly'
-            ],
+    levels: [
+        'log',
+        'off',
```
Comment (mdumandag) [inline code_samples/logging.js]: `off` should not be there

File: code_samples/logging.js [hunk 4]
```diff
@@ -19,59 +19,55 @@ var Config = require('hazelcast-client').Config;
...
 
-            levels: [
-                'error',
-                'warn',
-                'info',
-                'debug',
-                'silly'
-            ],
+    levels: [
+        'log',
```
Comment (mdumandag) [inline code_samples/logging.js]: `log` is not a `LogLevel`

File: code_samples/logging.js [hunk 5]
```diff
@@ -19,59 +19,55 @@ var Config = require('hazelcast-client').Config;
...
+        'off',
+        'error',
```
Comment (mdumandag) [inline code_samples/logging.js]: ordering of these should follow the order of the `LogLevel`
For example, `debug` must have the index of 3 (value of `LogLevel.DEBUG`)

File: code_samples/logging.js [hunk 6]
```diff
@@ -19,59 +19,55 @@ var Config = require('hazelcast-client').Config;
...
-                this.logger.log(this.levels[level], objectName + ': ' + message, furtherInfo);
-            },
+    log: function (level, objectName, message, furtherInfo) {
+        this.logger.log(this.levels[level], objectName + ': ' + message, furtherInfo);
+    },
 
-            error: function (objectName, message, furtherInfo) {
-                this.log(LogLevel.ERROR, objectName, message, furtherInfo);
-            },
+    off: function (level, objectName, message, furtherInfo) {
```
Comment (mdumandag) [inline code_samples/logging.js]: `off` is not a method of the `ILogger` interface

**Comment by mdumandag** [comment]
verify

File: README.md [hunk 7]
```diff
@@ -3327,12 +3327,18 @@ After enabling the client statistics, you can monitor your clients using Hazelca
...
-
+ By default, Hazelcast Node.js client uses a default logger which logs to the `stdout` with the `INFO` log level. You can change the log level using the `'hazelcast.logging.level'` property of the `ClientConfig.properties`.
+ 
+Below is an example of the logging configuration with the `OFF` log level which disables logging.
+ 
+```javascript
+cfg.properties['hazelcast.logging.level'] = LogLevel.OFF;
+```
+ 
+ You can also implement a custom logger depending on your needs. Your custom logger must have `log`, `error`, `warn`, `info`, `debug`, `trace` methods. After implementing it, you can use your custom logger using the `customLogger` property of the `ClientConfig`
```
Comment (Serdaro) [inline README.md]: ... property of `ClientConfig`.

**Comment by mdumandag** [comment]
verify

**Comment by mdumandag** [comment]
verift

**Comment by mdumandag** [comment]
verify

**Pull request merged**